### ISSUE TITLE
Bazel: support updating docker digests/tags in container_pull

### ISSUE THREAD

**Issue by dicarlo2** [renovatebot/renovate#2905]
**What would you like Renovate to be able to do?**

I'd like Renovate to automatically update my docker dependencies declared in a bazel WORKSPACE file using [rules_docker's `container_pull`](https://github.com/bazelbuild/rules_docker#container_pull-1).

**Describe the solution you'd like**

Given

```
container_pull(
    name="<name>",
    registry="<docker_registry>",
    repository="<docker_repository>",
    digest="<image_digest>",
    tag="<image_tag>",
)
```

update `<image_tag>` and `<image_digest>` using the same behavior as described in https://renovatebot.com/docs/docker/

**Describe alternatives you've considered**

I'm not sure there are any.

**Additional context**

Note that `tag` is technically not required when `digest` is provided.

**Comment by rarkins** [comment]
Thanks again @dicarlo2, and can you give one "real" example as well? 

Would the logic be:
- If tag is present then fetch the (latest) digest for that tag and digest update the `container_pull` if necessary.
- If tag is present and semver-like then look for newer "versions" to update both tag and digest to?
- If no tag then fetch the digest for `latest` and do the same?

**Comment by dicarlo2** [comment]
Example:
```
container_pull(
    name="hasura",
    registry="index.docker.io",
    repository="hasura/graphql-engine",
    # v1.0.0-alpha31.cli-migrations 11/28
    digest="sha256:a4e8d8c444ca04fe706649e82263c9f4c2a4229bc30d2a64561b5e1d20cc8548",
    tag="v1.0.0-alpha31.cli-migrations"
)
```

In terms of the logic, that sounds right, though I suppose it should be configurable similar to https://renovatebot.com/docs/docker/.

**Comment by rarkins** [comment]
Summarising requirements and plans:

This will be a new bazel "depType" for us (`container_pull`).
`name` will map to `depName`
`purl` will be constructed from `registry` and `repository` combined
`tag` will map to `currentValue`
`digest` will map to `currentDigest`
We will ignore the comment line (` # v1.0.0-alpha31.cli-migrations 11/28`) and not be responsible for keeping it updated

@dicarlo2 I see one "problem" already, thanks to your example. As you're probably aware, Docker tags only sometimes follow semver, and even then they still half-break it because of the need to include platforms in the tag. e.g. you might have a tag `8.14.0-alpine` where "-alpine" is not meant to be part of the semver tag but instead indicates the platform, and therefore Renovate only looks for semantic versions with the same suffix/platform.

In the example above, it looks like a "true" semver tag, e.g. you probably want us to tags like `v1.0.0-beta1.cli-stuff`? OR instead should we only match ones ending in `.cli-migrations`? In which case I think that is a type of versioning convention that satisfies no standard.

**Comment by rarkins** [comment]
In other words, I'm not sure of any standardised versioning that knows to upgrade from `v1.0.0-alpha30.cli-migrations` to `v1.0.0-alpha31.cli-migrations` but NOT to `v1.0.0-alpha31`

**Comment by dicarlo2** [comment]
@rarkins best-effort works for me. I would've been very surprised if the tag was updated correctly in this particular case, sorry for the poor example. Still, just knowing that there is (probably) an update would be very useful.

**Comment by rarkins** [comment]
I just implemented customisable version scheme in #2945. e.g. now you can define rules like:

```
  "packageRules": [{
    "packageNames": ["hasura/graphql-engine"],
    "versionScheme": "semver"
}]
```

The above configuration means that the default "docker" versioning approach (matching suffixes such as `-alpine`) gets replaced with regular "semver" versioning just for `hasura/graphql-engine`. So now your example is mostly achievable too.

**Comment by dicarlo2** [comment]
Awesome!!! Thanks @rarkins @Dmitry9 !
